src/docs/asciidoc/index.adoc

```diff
@@ -2030,6 +2030,81 @@ NOTE: Server generation will generate interfaces that you have to implement in o
 
 Please refer to link:api/io/micronaut/gradle/openapi/OpenApiServerSpec.html[OpenApiServerSpec] for the whole list of client configuration options.
 
+== Source Generator from Micronaut JSON Schema
+
+This plugins integrates with https://micronaut-projects.github.io/micronaut-json-schema/latest/guide/[Micronaut JSON Schema] in order to provide support for generating beans from https://json-schema.org[JSON Schema].
+
+The plugin adds a link:api/io/micronaut/gradle/jsonschema/JSONSchemaExtension.html[jsonschema] extension to the `micronaut` configuration block.
+
+Some important notes on configuration settings:
+
+- Default generation language is JAVA which can be modified. However, only the languages supported by  https://micronaut-projects.github.io/micronaut-sourcegen/latest/guide/[Micronaut SourceGen] are accepted.
+- The generator supports all remote references that obey at least one of the pattern rules given in `acceptedUrlPatterns` parameter. This parameter is configurable, and it's default value is `["^https://.*/.*.json"]`.
+
+The full list of configuration properties can be found link:api/io/micronaut/gradle/jsonschema/JsonSchemaSpec.html[in the documentation].
+
+=== Generating From URL
+
+You can generate sources by configuring the extension via the `fromUrl { ... }` block:
+
+[source, groovy, subs="verbatim,attributes", role="multi-language-sample"]
+----
+micronaut {
+    ...
+    jsonschema {
+        fromUrl("http://localhost:${mockServer.port}/animal.schema.json") {
+            acceptedUrlPatterns = ["^http://localhost:.*"]
+        }
+    }
+}
+----
+
+[source, kotlin, subs="verbatim,attributes", role="multi-language-sample"]
+----
+micronaut {
+    ...
+    jsonschema {
+        fromUrl("http://localhost:${mockServer.port}/animal.schema.json") {
+            acceptedUrlPatterns.set(arrayOf("^http://localhost:.*"))
+        }
+    }
+}
+----
+
+=== Generating From Local Files
+
+You can generate sources by configuring the extension via the `fromFile { ... }` block:
+
+[source, groovy, subs="verbatim,attributes", role="multi-language-sample"]
+----
+micronaut {
+    ...
+    jsonschema {
+        fromFile(file("animal.schema.json")) {
+            lang = "java"
+            outputDirectory = project.getLayout().getBuildDirectory().dir("generated")
```

Comment:
```suggestion
            outputDirectory = layout.buildDirectory.dir("generated/json-schema")
```

(Using a subdirectory here since there may be other plugins writing to `generated`)